### PR TITLE
Move `conda.core.solve._get_solver_class` to `CondaPluginManager.get_solver_backend`

### PR DESCRIPTION
**Pull request by jezdez**
### Description

As a follow up to @jaimergp's comment in https://github.com/conda/conda/pull/11993#pullrequestreview-1177196553 this deprecates `conda.core.solve._get_solver_class` in favor of a method of the `CondaPluginManager`.

### Checklist - did you ...

- [ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

File: conda/core/solve.py [hunk 1]
```diff
@@ -48,21 +48,9 @@ def _get_solver_class(key=None):
...
-    if key in solvers_mapping:
-        return solvers_mapping[key]
-    else:
-        raise CondaValueError(
-            f"You have chosen a non-default solver backend ({key}) "
-            f"but it was not recognized. Choose one of: "
-            f"{', '.join(solvers_mapping.keys())}"
-        )
+    from conda.plugins import solvers
+    # TODO add deprecation warning
```
Comment (jaimergp) [inline conda/core/solve.py:52]: ^ Came here to say this :D

Comment (jezdez) [reply conda/core/solve.py]: Done in https://github.com/conda/conda/pull/12093/commits/939991a0b2ec294d9403d30e1971d37111232f4d.

File: conda/core/solve.py [hunk 2]
```diff
@@ -48,21 +49,13 @@ def _get_solver_class(key=None):
...
-        return solvers_mapping[key]
-    else:
-        raise CondaValueError(
-            f"You have chosen a non-default solver backend ({key}) "
-            f"but it was not recognized. Choose one of: "
-            f"{', '.join(solvers_mapping.keys())}"
-        )
+    warnings.warn(
+        "`conda.core.solve._get_solver_class` is pending deprecation and will be removed in a "
+        "future release. Please use `conda.base.context.plugin_manager.get_cached_solver_backend ",
```
Comment (travishathaway) [inline conda/core/solve.py:54]:
```suggestion
        "future release. Please use `conda.base.context.plugin_manager.get_cached_solver_backend "
```

This comma doesn't belong there, right? 😅

**Review by travishathaway — CHANGES_REQUESTED** [review]

File: conda/core/solve.py (hunk 2 above)
Comment (jezdez) [reply conda/core/solve.py:54]: Yep!

File: conda/plugins/manager.py [hunk 3]
```diff
@@ -9,24 +9,34 @@
...
+    get_cached_solver_backend = None
 
-    def __init__(self, project_name: str | None = None, *args, **kwargs):
+    def __init__(self, project_name: str | None = None, *args, **kwargs) -> None:
         # Setting the default project name to the spec name for ease of use
         if project_name is None:
             project_name = spec_name
         super().__init__(project_name, *args, **kwargs)
+        # Make the cache containers local to the instances so that the
+        # reference from cache to the instance gets garbage collected with the instance
```
Comment (jezdez) [inline conda/plugins/manager.py:32]: https://www.youtube.com/watch?v=sVjtp6tGo0g is a good summary of this problem. Or https://rednafi.github.io/reflections/dont-wrap-instance-methods-with-functoolslru_cache-decorator-in-python.html.

**Review by travishathaway — APPROVED** [review]

**Pull request merged**